pkg/azuredisk/controllerserver.go

```diff
@@ -894,13 +900,47 @@ func (d *Driver) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshotRequ
 		azureutils.SleepIfThrottled(rerr.Error(), azureconstants.SnapshotOpThrottlingSleepSec)
 		return nil, status.Error(codes.Internal, fmt.Sprintf("create snapshot error: %v", rerr.Error()))
 	}
-	klog.V(2).Infof("create snapshot(%s) under rg(%s) successfully", snapshotName, resourceGroup)
+	klog.V(2).Infof("create snapshot(%s) under rg(%s) region(%s) successfully", snapshotName, resourceGroup, d.cloud.Location)
 
 	csiSnapshot, err := d.getSnapshotByID(ctx, subsID, resourceGroup, snapshotName, sourceVolumeID)
 	if err != nil {
 		return nil, err
 	}
 
+	if location != d.cloud.Location {
+		klog.V(2).Infof("create snapshot location(%s) is different from disk location(%s)", location, d.cloud.Location)
+		if incremental {
```

Comment:
what if `incremental` is false?

Reply:
Due to we can only copy incremental snapshot to new region, if `incremental` is false, we will return log for location is different and create local snapshot.

Reply:
if so, if `incremental` is false, and location is different from current k8s cluster, then it should return error.

Reply:
ok, I return error before the local snapshot is created.